Readability.js

```diff
@@ -577,10 +577,12 @@ Readability.prototype = {
     this._clean(articleContent, "link");
     this._clean(articleContent, "aside");
 
-    // Clean out elements have "share" in their id/class combinations from final top candidates,
+    // Clean out elements with little content that have "share" in their id/class combinations from final top candidates,
     // which means we don't remove the top candidates even they have "share".
-    this._forEachNode(articleContent.children, function(topCandidate) {
-      this._cleanMatchedNodes(topCandidate, /share/);
+    this._forEachNode(articleContent.children, function (topCandidate) {
+      this._cleanMatchedNodes(topCandidate, function (node, matchString) {
```

Comment:
Yeah, if this is the only use we might as well switch it over to `_removeNodes` and remove `_cleanMatchedNodes`.

Reply:
I think `_removeNodes` won't work as-is, because it takes a nodeList, and all of the children need to be checked recursively here. We could either add a new `_getNodeChildren` method (like _getNodeAncestors) that gets all the children recursively, and then pass the result to `_removeNodes`, or we could just move `_cleanMatchedNodes` into the loop (but that seems kind of messy):

```
this._forEachNode(articleContent.children, function (topCandidate) {
      var endOfSearchMarkerNode = this._getNextNode(topCandidate, true);
      var next = this._getNextNode(topCandidate);
      while (next && next != endOfSearchMarkerNode) {
        if (/share/.test(next.className + " " + next.id) && next.textContent.length < this.DEFAULT_CHAR_THRESHOLD) {
          next = this._removeAndGetNext(next);
        } else {
          next = this._getNextNode(next);
        }
      }
    });
```

Reply:
Oh, gah, sorry for missing that. Yeah, then let's leave it as you have it in the patch. @PalmerAL can we still go with using the constant? You can probably just store a copy as a local var in the outer function.

Reply:
Sounds fine to me; I've added that in 3a2d258bbb2ae7f1efa02e569fa3a45b2cebff31.